### PR TITLE
fix: value handling of DynamicUnitInputNumberWithSlider

### PR DESCRIPTION
**Pull request by yomybaby**
**Changes:**

This PR fixes and enhances the `DynamicUnitInputNumberWithSlider` component by improving its value handling:

1. Update Input behavior (fix bug and improve UX):
  - set the controllable value always.
1. Updated slider behavior:
   - Set value to 0 and disable slider and hide tooltip when min value exceeds max value
  - Added `filterOutInvalidMarks` function to remove marks that fall outside the valid range(min~ max).
1. Removed dependency on external `isMinOversMaxValue` function from `ResourceAllocationFormItems`

Addtionally this PR adds new stories:  
   - "Greater min than max" to demonstrate behavior when min value exceeds max value
   - "Extra marks" to show custom marks on the slider
   - "Extra marks with greater min than max" combining both scenarios

**Rationale:**

These changes improve the component's robustness and flexibility, handling edge cases and providing better visual feedback to users.

**Effects:**

- Users will see more consistent behavior when min/max values are set incorrectly
- Developers can now add custom marks to the slider and test various scenarios using the new stories

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#2730** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2730?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/DynamicUnitInputNumberWithSlider.tsx [hunk 1]
```diff
@@ -71,9 +86,7 @@ const DynamicUnitInputNumberWithSlider: React.FC<
           max={max}
           units={units}
           // set value to 0mib when min value overs max value
-          value={
-            isMinOversMaxValue(_.parseInt(min), _.parseInt(max)) ? '0m' : value
-          }
+          value={value}
```
Comment (ironAiken2) [inline react/src/components/DynamicUnitInputNumberWithSlider.tsx:89]: Doesn't DynamicUnitInputNumber require separation for isMinOversMaxValue?
```suggestion
          value={isMinOversMaxValue ? '0g' : value}
```

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
The value seems to keep changing whenever I move the focus in and out of the `DynamicUnitInputNumber`.

**Comment by ironAiken2** [comment]
<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/lSyr8xXz1wdXALkJKzVx/d96bad30-b0c4-462c-a691-4b90a2c829d9.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/lSyr8xXz1wdXALkJKzVx/d96bad30-b0c4-462c-a691-4b90a2c829d9.mov">Screen Recording 2024-10-02 at 11.38.32 AM.mov</video>

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
The values of input and slider do not match.

**Comment by yomybaby** [comment]
There is a mistake. I change `parseInt` to `parseFloat`. Please check again.

**Review by ironAiken2 — APPROVED** [review]
LGTM

**Pull request merged**